modules/uac_registrant/registrant.c

```diff
@@ -824,6 +824,34 @@ static struct mi_root* mi_reg_list(struct mi_root* cmd, void* param)
 	return rpl_tree;
 }
 
+int run_compare_rec(void *e_data, void *data, void *r_data)
+{
+	reg_record_t *old_rec = (reg_record_t*)e_data;
+	reg_record_t *new_rec = (reg_record_t*)data;
+
+	if ((old_rec->state == REGISTERED_STATE) &&
+	    (str_strcmp(&old_rec->td.rem_uri, &new_rec->td.rem_uri) == 0)) {
+		memcpy(new_rec->td.id.call_id.s, old_rec->td.id.call_id.s,
+		    new_rec->td.id.call_id.len);
+		memcpy(new_rec->td.id.loc_tag.s, old_rec->td.id.loc_tag.s,
+		    new_rec->td.id.loc_tag.len);
+		new_rec->td.loc_seq.value = old_rec->td.loc_seq.value;
+		new_rec->last_register_sent = old_rec->last_register_sent;
+		new_rec->registration_timeout = old_rec->registration_timeout;
+		new_rec->state = old_rec->state;
+	}
+	return 0;
+}
+
+int run_find_same_rec(void *e_data, void *data, void *r_data)
+{
+	reg_record_t *new_rec = (reg_record_t*)e_data;
+	int i = (int*)data;
```

Comment:
@ovidiusas could you review this a bit - while looking at the [Travis log](https://travis-ci.org/OpenSIPS/opensips/builds/97166562) I noticed this warning, which actually may lead to a bug.

`data` is a pointer to an integer, that is passed to an integer. This may be a _huge_ value, that is used below to index the `reg_htable`.

My guess is that the line should be `int i = *(int *)data;`, but please give it a look too.